### ISSUE TITLE
[Discussion] Prepare for the upcoming upgrade to .NET 4.7.2

### ISSUE THREAD

**Issue by davidebbo** [Azure/app-service-announcements-discussions#37]
Discussion thread for https://github.com/Azure/app-service-announcements/issues/89.

**Comment by niemyjski** [comment]
@davidebbo Would it be possible to have an azure site extension for latest stable as well as pre release versions of .net  and .net core depending on the users selected deployment lane?

**Comment by davidebbo** [comment]
@niemyjski It's not something that can be done via site extensions. There can be only one version of .NET 4.x on a given VM (by design of .NET fx), so you can't have a given app select among several.

**Comment by Daniel-Svensson** [comment]
Maybe this could handled by a setting on the app service plan itself (webfarm) in the future. With the rtm of 4.7.2 I see no big reason to have that for this version since I expect an update to be imminent. Would like to hear if an estimated date has been set.

**Comment by davidebbo** [comment]
@Workshop2 likely a number of months, to let the initial wave of issues get ironed out. We've had problems in the past doing it too fast and breaking some customers due to .NET regressions that had not yet been found.

Are there specific 4.7.2 features that you are waiting for? Unless there is, there is no reason to target it.

**Comment by Workshop2** [comment]
Thanks for the quick response @davidebbo.

We have legacy webforms apps we'd like to take advantage of DI 👍

**Comment by davidebbo** [comment]
I see. Please keep comments coming on this thread. We will try the strike the right balance between limiting risk to existing apps, and enabling the new features. Note that we don't have the ability to deploy it on a per-app basis, so it will need to be part of a future global update.

**Comment by richkgit** [comment]
We may switch to containers or other deployment options so as to be able to leverage framework updates.  Between the DI updates zip performance, etc. we would need those for a current project updating a Web Forms app.

**Comment by mrahl** [comment]
We would like to see the update sooner rather than later. We have had many customer escalations where the root cause is the cache memory monitor bugs 531575 and 531576 (mentioned here https://github.com/Microsoft/dotnet-framework-early-access/blob/master/release-notes/build-3052/dotnet-build-3052-changes.md).

We know how to work around the bugs by configuration, but the problem is customers don't always realize what they are dealing with. They see stability problems (because the apps get killed over and over by proactive auto heal) and look in all the wrong places before even coming to us so we can connect the dots and suggest the workaround.

**Comment by davidebbo** [comment]
@mrahl there are some bad SqlClient related regressions in 4.7.2 (see https://github.com/Microsoft/dotnet/issues/749). There needs to be a round of fixes on top of it before it can be deployed to app service.

**Comment by rokonec** [comment]
Well - 4.7.2 has finally resolved package/dll hell of System.Net.Http
Upgrade to 4.7.2 is only way how to get rid of assembly conflict warning messages in out solution.
Guess we have to wait a little bit longer before we can leverage it in app services.

**Comment by smithkl42** [comment]
The System.* DLL hell issues are a huge problem for us - we've spent many, many hours fighting with binding redirects, and it seems like we need to fix everything again every time we update a nuget package. We're informed that all this is fixed in .NET 4.7.2, and would love to switch over to it, but can't until app services supports it.

**Comment by jonorogers** [comment]
I'm another one in the System.Net.Http binding redirect hell - app service support for 4.7.2 can't come quick enough, in my book.

Question; when an exception is thrown currently, Azure App Services reports the following: 

> Version Information: Microsoft .NET Framework Version:4.0.30319; ASP.NET Version: 4.7.2623.0

Does this mean it's already running 4.7.2, as it seems like it is?

**Comment by davidebbo** [comment]
@jonorogers it has 4.7.1. Not sure I understand what part of this version string made you think it has 4.7.2.

**Comment by jonorogers** [comment]
> ASP.NET Version: 4.7.2623.0

That part confused me, I glanced at the version string and saw 4.7.2

**Comment by jrummell** [comment]
I'm also waiting for System.Net.Http fixes. Upgrading to 4.7.2 was the only way I found to get around the following VSTS View compilation error. I'm getting this in all of my web api projects with the Microsoft.AspNet.WebApi.HelpPage installed (e.g. dozens of projects).

> Error CS0246: The type or namespace name 'MediaTypeHeaderValue' could not be found (are you missing a using directive or an assembly reference?)

So I can either upgrade to 4.7.2 to fix the VSTS build/release and not deploy to Azure (everything is in Azure), or publish manually from Visual Studio on a few very specific workstations that have the magic combination of .net, Visual Studio and pixie dust.

**Comment by davidebbo** [comment]
@mariuszdotnet Yes, still on track for September, and that will include ASE.

**Comment by TheDude72** [comment]
@davidebbo:  Would there be any issues if I were to target .net framework 4.7.2 in my app and run on the .net framework 4.7.1 in the App Service?  So far, I haven't run into any issues.

**Comment by davidebbo** [comment]
@TheDude72 I guess that would be fine, though if you're not using any new features it doesn't buy you much

**Comment by dilv0055** [comment]
May be not a right place to ask, but any plan for upgrading VSTS Hosted Agents .net framework version  to 4.7.2 from 4.7.1?

**Comment by davidebbo** [comment]
@dilv0055 indeed, different team. I suggest trying https://github.com/Microsoft/vsts-agent

**Comment by dkadmin** [comment]
Hi,

Can anyone confirm, if .net .4.7.1 will remain or will be deprecated after releasing .net 4.7.2?

**Comment by davidebbo** [comment]
@dkadmin all VMs will be upgraded to 4.7.2 as part of this change, so none will be running 4.7.1. But as stated in the announcement, this should not affect any deployed app unless you hit an unexpected 4.7.2 regression.

**Comment by dkadmin** [comment]
@davidebbo thanks for the details. I just want to confirm 1 more thing, as of now we can see 2 version available (v3.5 and v4.7)  in webapp under application setting tab. So will be there 1 more new tab or will it remain same.

Thanks in advance.

**Comment by davidebbo** [comment]
@dkadmin that will not change. 4.7 really means 4.7.x, and it's just that it will move from 4.7.1 to 4.7.2.

**Comment by sampath42** [comment]
Hi Davidebbo
This framework upgrade is only for app service? or is it applicable for the cloud services as well?
From your above comments, "all VMS will be upgraded to 4.7.2 as part of this change", does this mean all Azure VMs?

**Comment by davidebbo** [comment]
@sampath42 This announcement only relates to App Service. I have clarified the sentence above.

@dkadmin There should be zero impact, outside of the very small chance of running into a .NET 4.7.2 regression that affects your app (hence the suggestion to test ahead).

**Comment by dkadmin** [comment]
@davidebbo thank you so much..

**Comment by itshemant09** [comment]
Is that possible to check the feasibility whether this update .NET Framework 4.7.2  plan first in our pre-prod subscription, so that we will test our application/services thoroughly. @davidebbo

**Comment by davidebbo** [comment]
@itshemant09 that is not an option due to the nature of our deployments. But you can test on a separate VN with the new framework.

**Comment by itshemant09** [comment]
@davidebbo VN ???

**Comment by davidebbo** [comment]
Sorry, typo, *VM*.

**Comment by itshemant09** [comment]
@davidebbo Thank you so much :-)

**Comment by ESSAdeveloper** [comment]
Hi David,

On Azure Will this 4.7.2 be available as a choice among the other versions or will it be the only version available?

thanks
ESSADev

**Comment by davidebbo** [comment]
@ESSAdeveloper see answer to same question [above](https://github.com/Azure/app-service-announcements-discussions/issues/37#issuecomment-413600732).

**Comment by pYThMik** [comment]
Hello, do you know / can you precise the actual date that 4.7.2 will be available on app service ?
Thanks

**Comment by davidebbo** [comment]
@pYThMik  @Sethu2488 I have added a 'Frequently asked questions' section to the announcement to cover those points

**Comment by leonmeijer** [comment]
How to tell if we have been upgraded? Since the Azure outage in South Central US, our app service doesn't behave as before. We use Azure AD authentication and we now suddenly have a different claims set. If we deploy the same app to West Europe, all works fine.

**Comment by davidebbo** [comment]
The upgrade to 4.7.2 has not yet started anywhere, so any issue you are having is definitely not related to it.

**Comment by mmulhearn** [comment]
Once the rollout begins, how can we determine if it has been installed on our apps?

**Comment by davidebbo** [comment]
You can use the following, based on [this page](https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed). From Kudu console, run:

    reg query "HKLM\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full" /v Release

- `0x709fe` -> .NET 4.7.1
- `0x70bf6` -> .NET 4.7.2

It has not yet started, and probably will not until Friday 14th at the earliest.

**Comment by dcc635** [comment]
To clarify, Microsoft is forcing all machines to be upgraded to 4.7.2, so we need to test it early to work out any potential bugs. But until they force the upgrade we can't actually test on production since 4.7.2 is not available in the Azure VMs yet?

Also, does "upgraded" mean all apps will be changed to target 4.7.2? Or just that the VMs will have 4.7.2 available for targeting?

**Comment by mmulhearn** [comment]
@dcc635 What we did was spun up a VM, manually installed 4.7.2, set up our site in IIS and tested that way.  There were a few bugs that were attributed to this quick set up, but we got the app tested to ease people's minds.

**Comment by tjrobinson** [comment]
Thanks for the update @davidebbo 

We are consuming .NET Standard 1.5+ from .NET Framework which is "broken" prior to .NET 4.7.2. See: https://twitter.com/terrajobst/status/1031999730320986112 and https://github.com/dotnet/docs/issues/7305 among others. So we're very keen for the upgrade.

Why the wait until after Ignite? Is something being announced perhaps? Or is it just lack of resources until then?

A little off-topic, but do you know if .NET 4.7.2 is coming to Cloud Services anytime soon?

**Comment by davidebbo** [comment]
> Why the wait until after Ignite? Is something being announced perhaps? Or is it just lack of resources until then?

We've had delays deploying the *previous* App Service monthly update (not containing 4.7.2). And now we wanted to limit the churn just before the conference, so we decided to move it to after. So it is somewhat related to Ignite, but not tied to any specific announcement.


> A little off-topic, but do you know if .NET 4.7.2 is coming to Cloud Services anytime soon?

Sorry, not my area so I really do not know. I suggest asking on the forum for that service.

**Comment by HowardHoffman** [comment]
Request for clarification:  Will a WebJob built with .NET 4.7.2 start successfully in the AzureAppService prior to October?  We attempted to test in Azure in a non-production AppService and our (continuous) WebJob stops and sits at 'Pending Restart' status.  Backing the change out and switching to 4.7.1 resolved the issue.  This expected?

**Comment by davidebbo** [comment]
@HowardHoffman This is more of a .NET question than an App Service or WebJobs question. If you run an app targeting 4.7.2 on a VM that has 4.7.1, I would not expect it to run.

**Comment by tusmester** [comment]
Hi @davidebbo , I have a strange question. A couple of weeks ago I have deployed an app to App Service and _it has been working_ for at least a few days. I'm pretty sure (99%) that the app is not able to work without .Net Framework 4.7.2 present in the environment (due to a [type forwarding issue](https://github.com/dotnet/standard/issues/300), that was fixed in that version). But now it does not work anymore and it throws that same ReflectionTypeLoadException that I'm familiar with from our local environment and which can be solved by installing 4.7.2.

So I'm curious, is it possible that _4.7.2 was actually deployed_ at some point on at least some of your machines and than a rollback happened? I checked the currently installed version on the portal using Kudu and it is 4.7.1, which explains the error but it does not explain why the app was working a few weeks ago.

(to be precise: I deployed the app, it was working, I did not touch it or check it for a couple of weeks, than **I published a new version** which does not work - but the changes were not related to the .net framework version in any way)

So my argument is not that it should be working now but that the app **should not have been working before** :).

**Comment by davidebbo** [comment]
@tusmester I really can't explain that. No VM in App Service should have ever had 4.7.2 on it.

**Comment by kdubau** [comment]
>(to be precise: I deployed the app, it was working, I did not touch it or check it for a couple of weeks, than I published a new version which does not work - but the changes were not related to the .net framework version in any way)

Same thing happened to me :/ 

My app has targeted 4.6.1 for over a year, has always worked in AppService (and continues to work). Today I pushed a super minor code fix to my staging slot, and suddenly a ton of exceptions in System.Net.Http... I didn't change anything related to build/references/etc

```Could not load file or assembly 'System.Net.Http, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.```

Not sure what to do here as it's quite hard to debug since it only happens on AppService (local has no issues).

@davidebbo that makes at least two reports of "something changed on AppService" that broke my app. Do you have a list of change made to the AppService VMs in the last few months? Anything would be helpful..

**Comment by dmarlow** [comment]
@kdubau Take a look here regarding System.Net.Http https://github.com/dotnet/corefx/issues/29622

That was related to my efforts in upgrading to 4.7.2 to find out App Service didn't yet support it: https://github.com/Azure/app-service-announcements-discussions/issues/37#issuecomment-391884624

**Comment by kdubau** [comment]
In my case, I've always target 4.6.1 and still do but suddenly things don't work even though we are told AppService is still on 4.6.x

**Comment by davidebbo** [comment]
@kdubau This thread is specifically about discussing the upcoming deployment of 4.7.2, which has not started. So any issue you are seeing is almost certainly unrelated to it. Doesn't mean we shouldn't discuss your issue. But I just don't want to do it here as it's going to cause confusion if we discuss unrelated things. You can use either StackOverflow, or the Web Apps MSDB forums. But you'll need to provide more details, and ideally a complete minimal runnable repro.

**Comment by Jehoel** [comment]
@davidebbo 

> If you run an app targeting 4.7.2 on a VM that has 4.7.1, I would not expect it to run.

I just deployed an ASP.NET Core 2.0 project targeting .NET Framework 4.7.2 to an Azure AppService running 4.7.1 and my application seems to be running fine so far - I'm not sure what to make of that - but isn't the .NET Framework meant to be forwards-compatible anyway? (i.e. the Fx4.7.1 will try its best to run a 4.7.2, 4.8, 4.9, etc application?). FWIW my project doesn't use any new API surfaces introduced in 4.7.2.

**Comment by davidebbo** [comment]
@Jehoel Yes, you're probably right, if it doesn't use any new features. In any case, the main point here is that right now App Service is running 4.7.1, so the behavior you get should be no different than what you would see on any VM with 4.7.1.

I'm trying to keep the discussion focused on concerns that are App Service specific, as opposed to discussing the behavior of the .NET framework in general.

So to summarize:

- While App Service has 4.7.1, it should behave like any other VMs with the same version
- Once it gets 4.7.2, it should  behave the same as any other VM with 4.7.2

What's worth discussing here are cases where users encounter situations that go against those last two statements.

**Comment by mamoorkhan** [comment]
Now that [Microsoft Ignite](https://www.microsoft.com/en-us/ignite) is over, can we please get an update on the Regions where the deployment will be happening and when will it be happening, we are specifically interested in the dates for **North Europe** as we have quite a few App Services in there hanging in the limbo.

**Comment by egzonzeneli** [comment]
@davidebbo is there an ETA for CUS region ... or will be delayed again?

**Comment by davidebbo** [comment]
@egzonzeneli We're targeting to complete the deployment by the end on October. It still has not started.

**Comment by Richard004** [comment]
Ok. It is good to know. I am working on a feature that should be ready by the end of October. The point is that the branch where I work is using .net standard library with [OnDeserialized] attribute. That fails on net471, but is fixed on net472. Fixed issue (https://github.com/dotnet/standard/issues/300)
Is there a chance that I will be able to release it by the end of October on Azure app service?

Of should I better start refactoring the [OnDeserialized] out to meat the deadline?

**Comment by davidebbo** [comment]
> Is there a chance that I will be able to release it by the end of October on Azure app service?

Yes, there is a chance that you'll be able to, but it's possible that the worldwide deployment completion will leak into early November if there are delays. So it all depends if you have a hard deadline on November 30th.

**Comment by Richard004** [comment]
Thank you David.
This is good enough estimation for me.
I'll make the decision at about 25.10.
Your info helped.

**Comment by davidebbo** [comment]
> So it all depends if you have a hard deadline on November 30th

And to clarify, this should have been:

> So it all depends if you have a hard deadline on **October 31st**

**Comment by jeffbdye** [comment]
@davidebbo Sounds like this has been confirmed to be pushed back and 4.7.2 won't be available until the end of October? If so, probably want to get the announcement here updated since it's still saying mid October: https://github.com/Azure/app-service-announcements/issues/89

**Comment by davidebbo** [comment]
@jeffbdye it's meant to be *starting* in mid-October, and going through the rest of the month, and that is still the plan. But it starts slow and accelerates, so most of it is in the tail end.

**Comment by deanswiatek** [comment]
.NET Framework 4.7.2 resolved my System.Net.Http issues that I could not get working on 4.7.1 at all. It cannot get deployed to Azure App Services fast enough as I've been waiting to move a couple of apps off of a VM to App Services, but can't until 4.7.2 is deployed.

**Comment by davidebbo** [comment]
FYI is you want to do some early testing, the West Central US region now has the new build.

**Comment by rkokXXVV** [comment]
Hi guys. I just tried to deploy a webjob to azure app service build using 4.7.1 and the application keeps crashing as soon as it starts up. Even tried with 4.7 and 4.7.2 . The application only ran when it was compiled using .net 4.6.1. I'm confused here as the settings page on azure web job clearly ask for either 3.5 or 4.7.

Thanks for the advise

**Comment by davidebbo** [comment]
@rkokXXVV Are you trying this on a VM that has been upgraded to 4.7.2 or not? See https://github.com/Azure/app-service-announcements/issues/89 for instructions on figuring this out. If it only happens on upgraded VMs, we should discuss here. If on the other hand you are seeing this on a VM with 4.7.1, then it is not related to this upgrade and is out of scope for this thread (e.g. you may want to start a StackOverflow question).

@kim3er for large regions with many scale units, they're not done all at one. So there is not really a region ordering in that sense.

**Comment by DarinMacRae** [comment]
Hi @davidebbo , your efforts on the roll-out of this upgrade are much appreciated. Addressing the System.Net.Http versioning issue alone is well worth it for me.

Sorry if you've said this somewhere before but is there a place we can monitor the roll-out, say, per region?

**Comment by MisinformedDNA** [comment]
@davidebbo I didn't think to check if Azure had the latest .NET, so I already pushed. Doh! I'm in North Central US, any idea if that's getting rolled out to today? Otherwise, I need to go fix all my references...

**Comment by MisinformedDNA** [comment]
Also, I don't have access to see what version is currently running, so I'm not even 100% sure that the framework is the problem.

**Comment by davidebbo** [comment]
@DarinMacRae See my previous comment about the roll-out of larger region, which doesn't happen all at once. So there is no per-region schedule per se. It will happen progressively in the next few days (might go into Nov 1 or 2). I know it's taking a long time, so please be patient, it'll get there soon :)

@MisinformedDNA see https://github.com/Azure/app-service-announcements/issues/89 for simple instructions on determining which version is on your VM.

**Comment by MisinformedDNA** [comment]
@davidebbo Like I said, I don't have access, but I finally got in touch with operations and sure enough, we don't have the update. :(

**Comment by Mikkelcv** [comment]
One of our sites has received this update and now we experience an altered behavior related to Application Settings when swapping slots.

If you have an Application Setting on "warm-up" slot, that is marked with "Slot setting" set to true and this setting does **not** exists on the Production slot, then during swap this setting is applied to the Production slot also.
This is not the case in the previous version.

Example from Azure portal on the "warm-up" slot:
![image](https://user-images.githubusercontent.com/39376179/47785130-077e6180-dd08-11e8-9603-f27690dec430.png)

This was not a nice surprise that all our web jobs no longer is running in production.

I can create the same Application Setting on production and assign it to 0, then it is not overwritten. But the documentation https://docs.microsoft.com/en-us/azure/app-service/web-sites-staged-publishing does not tell that this is the case.

Thanks
Mikkel

**Comment by davidebbo** [comment]
@Mikkelcv note that the update contains more than just the new .NET Framework, and what you report is almost certainly unrelated to it. Could you please start a thread on the [Web App forum](https://social.msdn.microsoft.com/Forums/azure/en-US/home?forum=windowsazurewebsitespreview), so we can properly discuss it? I will ask the right people to look at it asap.

**Comment by Mikkelcv** [comment]
Thank you, I have posted it here:
https://social.msdn.microsoft.com/Forums/azure/en-US/f2824e37-7b8f-4acc-b271-bb5c9c970aff/the-option-quotslot-settingquot-is-no-longer-obeyed-for-application-settings-which-is-only?forum=windowsazurewebsitespreview

**Comment by Daniel-Svensson** [comment]
Just noticed that the announcement was closed, does this mean that the rollout has completed and its available everywhere?

Would be nice with a statement saying so in https://github.com/Azure/app-service-announcements/issues/89

**Comment by xt0rted** [comment]
None of my VMs in US East are on 4.7.2 yet so I don't think the rollout is finished.

**Comment by mmulhearn** [comment]
Agreed; none of my web apps have been upgraded.

**Comment by fleetr** [comment]
None of mine either. (West Europe)

I think a lot of people would appreciate some sort of progress update/estimation, please?

**Comment by mmulhearn** [comment]
My mistake; my West US installation is on 4.7.2.  Central US and East US is not.

**Comment by markheath** [comment]
I've seen 4.7.2 in North Europe and West Europe, not on UK West yet

**Comment by davidebbo** [comment]
@RonniePitts not related to .NET 4.7.2, but possibly related to other things in the same update. Please check if you are seeing [this](https://social.msdn.microsoft.com/Forums/azure/en-US/f2824e37-7b8f-4acc-b271-bb5c9c970aff/the-option-quotslot-settingquot-is-no-longer-obeyed-for-application-settings-which-is-only?forum=windowsazurewebsitespreview), and if so, let's take the discussion there (issue should now be fixed).

**Comment by RonniePitts** [comment]
Thanks David.

Cheeky update then, at least once all the firefighting is done we'll have 4.7.2 plus some other features.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by snekbaev** [comment]
seems like my NE WebApps are running on 4.7.2 now

**Comment by MisinformedDNA** [comment]
I'm still on 4.7.1. @davidebbo Is the release still on-track to be finished by EOD?

**Comment by Daniel-Svensson** [comment]
our apps in North Europe are updated to 4.7.2 but WE is still 4.7.1

**Comment by markheath** [comment]
Today it seems all our UK West are on 4.7.2, but in West Europe and East US, there is a mix of 4.7.2 and 4.7.1

**Comment by Richard004** [comment]
I see that our West Europe app service plans are on 4.7.2 and all seems to work as expected. 
Big thanks to the team.

**Comment by MisinformedDNA** [comment]
I'm still on 4.7.1. @davidebbo What's the status?

**Comment by davidebbo** [comment]
It's on the last stretch now, and most app should already be upgraded. @MisinformedDNA can you check again for your app in North Central US? That region should be almost completely upgraded now.

**Comment by penfold** [comment]
@davidebbo I'm on UK West with 4.7.1 - do you have an ETA for this region/zone to complete? Will the changes occur at any time or is it out-of-hours/over-night?

Thanks, Paul

**Comment by mariusGundersen** [comment]
@davidebbo what is the status for ASEs with in vnets? I have webapps in North Europe and they have not been updated yet, according to the registry value.

**Comment by davidebbo** [comment]
@mariusGundersen Indeed, ASEs have a slightly different completion schedule which will stretch until early next week.

By now, pretty much everyone else not using an ASE should have been upgraded.

**Comment by MisinformedDNA** [comment]
Finally got it!

**Comment by snekbaev** [comment]
@davidebbo Is it safe to update apps to 4.7.2 now? Just asking that things won't be reverted to 4.7.1 in the last minute :)

**Comment by davidebbo** [comment]
@snekbaev if you have been upgraded to 4.7.2, it is safe to assume it won't be reverted.

**Comment by kstreith** [comment]
Is the migration complete yet? Is there an ETA for completion? Particularly interested in West Europe.

**Comment by MisinformedDNA** [comment]
> By now, pretty much everyone else not using an ASE should have been upgraded.

@kstreith Are you using on an ASE?

**Comment by davidebbo** [comment]
I would expect 4.7.2 to be everywhere that's non-ASE by now. But I've now moved on to other things, so I'll let others give an official status. @stefsch, can you help with this? Thanks!

**Comment by czb182** [comment]
When can we expect this to be completed for ASE environments?

**Comment by danpetitt** [comment]
What version IS actually installed on ASE v2? There is no easy way to find out, is it v4.7, v4.7.1 or v4.7.2

**Comment by mariusGundersen** [comment]
@danpetitt it is explained in the announcement how to find out: https://github.com/Azure/app-service-announcements/issues/89

**Comment by jahmai** [comment]
What's going on with this? My service fabric nodes in UKW, WUS, AEP are yet to be upgraded.

**Comment by fabiocav** [comment]
Closing this discussion as the deployment is complete.

@jahmai this announcement and deployment covers App Service (Web Apps and Azure Functions), for Service Fabric information, I would recommend the MSDN forums [here]([url](https://social.msdn.microsoft.com/Forums/en-US/home?forum=AzureServiceFabric))
